### ISSUE TITLE
Transparency concerns.

### ISSUE THREAD

**Issue by mikeal** [nodejs/inclusivity#8]
I'm assuming that the charter proposal, https://github.com/nodejs/diversity/pull/7, will pass and so I'm addressing these concerns in the context of the responsibilities stated in that document.

For reference, those goals are:

* Foster a welcoming environment that ensures participants are valued and can feel confident contributing or joining discussions regardless of their background or level of experience
* Proactively seek and propose concrete steps the project can take to increase
inclusivity.
* Acknowledge and celebrate existing diversity accomplishments within the
project while seeking to build upon them.
* Identify ways to measure diversity and inclusivity within the project. Collect
the measurements and report on them at regular intervals.

These are all great, and can be accomplished across the entire project (Core and all WGs) and against additional top level projects in the future. I think we need a WG leading this charge and I'm happy to see some movement.

However, I believe the current "private" forum used for WG discussion is a huge barrier to getting those done. Cross collaboration across WGs is common in the project and this happens pretty easily because the same tools (GitHub) are used across the project and everyone has visibility in to what is going on. We've barred private forums in the project with the exception of security issues.

Transparency is becoming one of the keystones to the success of the re-formed project. In fact, the 3 requirements for all projects coming in to the foundation is to be [participatory, effective, and transparent](https://github.com/nodejs/TSC/pull/2/files#diff-4d0d41696abfdf5519ac03713216773dR28).

My understanding of the reason behind this particular forum *not* being transparent is concerns about privacy. However, we can't actually defend, legally speaking, the privacy of the conversations that are happening there and if any of the individuals involved, or the foundation, were to be sued all of what has been said there is discoverable. If privacy or anonymity is a concern then we need to find a way to provide it that we can actually protect.

I expect that what works for this WG will not be identical to other WGs, that's why the governance structure we have doesn't impose a particular set of processes across all WGs but I do think this WG will need to find a new system that achieves a level of transparency that doesn't inhibit it from being effective across the project while accommodating some privacy/anonymity when necessary in a way that we can truly protect.

**Comment by sup** [comment]
another thing to add, I've been interested in being part of this WG for a longer time now, and I've gotten some positive replies (I think), but in the end, nothing really happened. I understand that there has to be sort of an 'evaluation process' to prevent any random person from becoming part of the WG, but I would've at least expected a note of acceptance/denial, or maybe more _transparency_ overall.

**Comment by nebrius** [comment]
Hi @sup. I'm really sorry about this.

As you say, there needs to be a process in place, but unfortunately there isn't _any_ process in place yet. This has nothing to do with how we evaluated you or anything like that, because to be completely honest, we don't even know _how_ to evaluate people, much less the specific processes for evaluating people.

I recommend dropping by https://github.com/nodejs/diversity/pull/7 and leaving your thoughts so we can hopefully start to get this implemented.

**Comment by ashleygwilliams** [comment]
hey ya'll, weighing in here with my two cents:

- this WG, its charter, its members should be publicly avail on github AND the node site
- there should be a private channel for dealing with incidents
- reporting incidents should be by a private method
- public notes should be published, anonymously summarizing the above 2 pts
- a public channel for discussions should exist in parallel to the private channel, should be consistently moderated
- issues (and this repo), should be public, consistently moderated
- code of conduct should be one click away for all of these

**Comment by juliepagano** [comment]
+1 to @ashleygwilliams on having a mix of private and public spaces for the working group. Not all issues it will address are best solved in a super-public venue.

**Comment by mikeal** [comment]
> this WG, its charter, its members should be publicly avail on github AND the node site

Another thing to keep in mind is that membership in the GitHub org also all the members free registration as individual members in the Node.js Foundation which gives you voting rights for the election in January. If you want to run for the board and/or have voting rights that will be important.

**Comment by rvagg** [comment]
> * there should be a private channel for dealing with incidents
> * reporting incidents should be by a private method

If this is going to be part of what this WG does then it needs to be absolutely clear for what purpose there will be such discussions and what the goals will be coming from any closed-room discussions. We should not allow any perception of a Spanish Inquisition style body coming out of this. Any shred of moral authority and respect will quickly drain away, particularly if membership of this group has little overlap with the rest of the org.
My fear is that if this mechanism is officially in place then it'll easily become the default method of communication, there will be little transparency and the group's existence will be pointless and not lead to any concrete outcomes.

**Comment by ashleygwilliams** [comment]
hi @rvagg. this is a very aggressively communicated, premature concern. the suggestion that handling incidents in private is akin to the "Spanish Inquisition", is well.... completely inappropriate. we will be having a meeting to discuss this. i have been a part of many groups that balance public and private means of communication effectively.

if you are interested, please consider attending to listen to our plan and voice further concerns (perhaps more constructively than this.)

**Comment by jasnell** [comment]
@ashleygwilliams, @rvagg raises a valid concern that should be considered. I don't see it as being premature or inappropriate at all, and it definitely should not be dismissed out of hand. As I read Rod's comment, I see no indication that he is equating "handling incidents in private" as being akin to the "Spanish Inquisition" -- the comment, as I interpret it, Rod is voicing a concern that is intended only to encourage as much transparency and openness in the process as is practically feasible so that this WG can succeed in it's goals.

Having a mix of private and public discussions is perfectly valid. The TSC and CTC meetings themselves are clearly divided into private and public sections, as are the board meetings. The purpose of the private portions are to discuss potentially sensitive matters. Duplicating that pattern for this WG and accounting for private conversations to happen is simply the right thing to do. No one is arguing against that or even suggesting in any way that such private conversations are inappropriate. However, if (a) the purpose and intent of those private conversations begins to expand and (b) this WG too aggressively attempts to impose enforcement of CoC related matters without actively seeking buy-in from rest of the community, then this WG will end up undermining itself and it's goals (rightfully or wrongfully).

**Comment by ashleygwilliams** [comment]
i think the concern is valid, the delivery hyperbolic. a couple of things:

1. "membership of this group has little overlap with the rest of the org" suggests that members of this group will not be empowered to participate more fully in the community and already begins to alienate the group. while it is part of this group's jobs to get buy in from the TSC, so it is the TSC's community to actively empower and integrate this team.

2. "My fear is that if this mechanism is officially in place then it'll easily become the default method of communication, there will be little transparency and the group's existence will be pointless and not lead to any concrete outcomes." fears are fears and sharing them is good but the language here is aggressive and shows very little faith in the team, which is p harmful at such an early stage in the process.

**Comment by ashleygwilliams** [comment]
tl;dr

let's all agree to be constructive, if you have fears + concerns share them. i trust everyone here cares about this being successful, but sometimes it's hard to hear that care in comments.

my fifth grade teacher once said "don't present a problem without a suggestion of a solution". to me this is a p good (tho quite short) definition of constructive criticism. and even if you are not sure of how to solve something, spend some time in your comment speculating on good questions we need to answer or at least components of what might help solve the problem. at the VERY least, acknowledge that the problem is Hard and you want to help solve it.

also: there *is* a power dynamic here as well. let's all agree to be aware of that.

**Comment by jasnell** [comment]
@ashleygwilliams ... there are several TSC/CTC members here that are already *actively* engaging in the conversation, providing feedback, voicing concerns and helping with the process as much as possible. I do not see anything in @rvagg's comments that indicate any doubt in the abilities of this group to accomplish it's goals, explicitly stated or otherwise.

**Comment by ashleygwilliams** [comment]
if i haven't already explained it to you, perhaps i won't be able to communicate it to you. i'm coming from a different place than both of you tho, so that's work considering.

again, my main point is let's keep contributions constructive and not hyperbolic.

**Comment by jasnell** [comment]
While I do not believe it was the intention, the comment "*if i haven't already explained it to you, perhaps i won't be able to communicate it to you. i'm coming from a different place than both of you tho, so that's worth considering*" is actually fairly dismissive and counter to what I believe the goal to be. There is certainly the possibility that we are misunderstanding one another, but my goal here is to *be* constructive: We simply cannot foster an inclusive, diverse community if we are at all dismissive of the concerns of the very people we are trying to influence and guide.

**Comment by nebrius** [comment]
Hey everyone. I've said this a few times before, but it bears repeating:

This WG will not have enforcement powers over other groups.

Any enforcement power over other groups will come from the TSC or CTC, if not the foundation directly.

The purpose of this group is to formulate plans and ideas that the TSC/CTC can implement in a wider context.

Directly giving this group any sort of enforcement power over other groups is not on the table.

It will probably be the case that the enforcement body, whatever it looks like, will be made up of some members from this group, using ideas generated by this group, but both will still need to be approved by the TSC/CTC first.

Think of this group like a sub committee in the US Senate. Legislation often starts there, but they do not pass legislation by themselves.

**Comment by jasnell** [comment]
@nebrius ... that is certainly the right approach, I think. I would take it a step further and say that the TSC should invite a representative from this WG to participate in the TSC meetings as either an observer or full voting member. Doing so will ensure that this WG has clear representation and visibility to the entire project, including any potential discussions involving enforcement.

**Comment by mikeal** [comment]
I do think that we're hoisting a bit too much criticism on an un-chartered group that is preparing for their first meeting. That said, I think that some of the underlying context below these concerns can help this group create a better draft charter and hopefully produce a more productive meeting.

While most people assume the project is built as a strict hierarchy, it is not. The TSC is responsible for all technical matters in the foundation but as it spins up Top Level Projects it delegates authority and autonomy to that project. That project in turn creates working groups which, once chartered, are also given autonomy over the responsibilities in their charter. The premise here being that the contributors doing the work in these groups have ownership and freedom without being "bossed" by another group in a hierarchy. Rather than looking like a company or traditional institution it's more of an anarchist web of responsibilities and service providers that are building complimentary competencies.

Now, an argument has been made, which much evidence, that not all of these groups contain a core competency for matters of inclusivity. It makes sense then that we would want to delegate authority and autonomy for that competency to this group, even though it is not yet chartered and has not actually asked yet for these responsibilities to be delegated to it. However, people have already [asked that](https://github.com/nodejs/inclusivity/issues/2) matters like enforcement be delegated.

Delegating that *is* going to be difficult, not because the TSC or CTC won't support it or can't pass it, but because once autonomy is granted to all of these groups the TSC and CTC traditionally maintain very little authority in them and as a result don't have to maintain much of a mandate. To Rod's comment that this group won't have a ton of overlap with the rest of the org, that is true of *every* working group, the members of which have limited overlap with the rest of the org. Even the CTC has members that overlap maybe 30% with the org as a whole, which is one of the reasons we have several people attend the meetings who don't have voting privileges. Rod wasn't saying that this group will somehow inherently be less involved with the rest of the org, he's just expecting this group is as involved as any other WG, which is not much because of the size and scope of the org.

Obviously you are still designing the process by which the group will operate. As you do this keep in mind that while *authority* for this task will only take a vote from the TSC, maintaining a community *mandate* across the org is much more difficult and the process which you create and maintain for it will have a role in that.

**Comment by ashleygwilliams** [comment]
to reiterate: this has been a very not inclusive discussion for me, a member of an underrepresented group for whom this WG is supposed to be focused on helping to include.

i'm going to keep being here. keep calling out shit. keep working to build supportive procedures, effective workflows, etc. but heads up? this is where you lose a LOT of people. for those that even dare join? you lose them *RIGHT NOW*.

care about this. and czech yourselves, ya'll. this is my final comment for now, i need to go get some work done.

**Comment by nebrius** [comment]
> @nebrius ... that is certainly the right approach, I think. I would take it a step further and say that the TSC should invite a representative from this WG to participate in the TSC meetings as either an observer or full voting member. Doing so will ensure that this WG has clear representation and visibility to the entire project, including any potential discussions involving enforcement.

I think that's a really good idea, inviting someone to the TSC. I recommend someone from an underrepresented group that's involved in the WG, such as @ashleygwilliams or @juliepagano. Do either of you have the desire/time to become a TSC member?

**Comment by ashleygwilliams** [comment]
i'd like more details on the exact time/effort commitment and an agreed upon "role definition" (even a vague one to start with would be :ok_hand: ), but overall, my response is yes.

**Comment by nathan7** [comment]
@rvagg @jasnell The standard already *is* for diversity/inclusivity efforts and under-represented groups to be met with scepticism and accusations of trying to take something away from the well-represented groups.
"Constructive" criticism like this isn't novel, it's already is the background noise against which *any* discussion of this subject takes place, and the fact that you're not taking this into account doesn't really convey any effort to empathise. And empathy is a thing we should be able to expect from people in positions of power in the project. Scepticism is plentiful already.

**Comment by mikeal** [comment]
@ashleygwilliams Since the TSC delegated project responsibility to the CTC it's actually very light as a time commitment, although I will warn that when we do have meetings they tend to be *incredibly* boring as they deal with a lot of the procedural requirements of maintaining the technical side of the foundation but don't include any of the actual code work. Right now the meetings are fewer than one every two weeks (we have a meeting once there is agenda that has been escalated to it, which isn't happening on a super consistent basis).

For some idea of what that looks like checkout https://github.com/nodejs/TSC

**Comment by nebrius** [comment]
@rvagg, @jasnell I appreciate your feedback that you've given. Getting the right balance between privacy and transparency is really hard. I don't think this comes as a surprise to anyone.

I would like to remind you that the topics this group deals with are different than what the others deal with. This isn't about technical discussions, which have something at least in the ballpark of a right and wrong answer. Here we're talking, ultimately, about the way people feel. I would urge you (and everyone else here) to try and be cognizant of that fact.

Pretty much everything said in this group will be emotionally charged. I ask that everyone please try and approach everything with more compassion, and patience, than in other threads. Be extra sensitive, because these discussions _need_ it. (I am including myself in this, as I haven't always been good at this).

I would also like reiterate what Mikeal said, we're still _very_ new. More so than most people realize as the group has been dormant most of it's existence. In practice, this group is the equivalent of about two weeks old.

**Comment by jasnell** [comment]
@ashleygwilliams ... In terms of "role definition", I believe that it would really come down to being a strong advocate for diversity/inclusivity related issues, ensuring that issues are raised and discussed adequately, raising the visibility of those issues within the community, and helping to ensure that any conflicts that do arise are handled appropriately in accordance to the project and foundation policies. 

Once this WG is officially chartered, it would take an existing TSC member to nominate the selected member of this WG to join as a new TSC member -- which I am more than happy to do as I believe that would be very important and key to helping this WG meet it's goals -- as well as key to addressing the concern that @rvagg raised.

**Comment by mikeal** [comment]
Something to keep in mind, if this WG wants representation in the TSC and wants to work on foundation-wide vs project-wide issues then it'll end up being a WG of the TSC and not the CTC. In fact, it would actually be the *first* top level working group.

**Comment by Fishrock123** [comment]
Fwiw I'm pretty sure this thread is quickly becoming too much to digest except for those currently involved.

**Comment by sup** [comment]
i agree with @Fishrock123, this thread's been having a lot comments. maybe we should sum the current state of discussion and some ideas up and put them in the OP?

**Comment by mikeal** [comment]
Since I created the issue I'll close it. This is addressing issues with how the group was operating in its infancy and I'm confident this will now be addressed by as the group creates a process and a charter going forward.

**Comment by rvagg** [comment]
Having my comment so wildly misinterpreted does not exactly inject me with confidence in the future of this WG.

**Comment by juliepagano** [comment]
Invoking the Spanish Inquisition, no matter how well intentioned, isn't a great way to produce productive dialogue from anyone. :-/

**Comment by Trott** [comment]
@rvagg This may be stating the obvious, but you are an extremely influential person in the project. For better or for worse, that puts you under a microscope. When you make an offhand mildly negative comment, it has Huge Impact. Especially on sensitive topics, if you don't take great care about what you say and how you say it, your comments will be misinterpreted. That's part of the package of being influential and privileged.

**Comment by isaacs** [comment]
@rvagg I know it's very frustrating and not at all fun to try to communicate something important to you, and see people misinterpret your words.

I've done a bit of work to try to figure out what you were trying to communicate above.  It's a bit scary to try to communicate this to you, because I'm telling myself the story right now that you're angling for a fight, and I'd rather not fight about this.

Several people have brought up the concern, based on their experience with this group and others, that reporting or discussing inappropriate conduct in public means that the victims of inappropriate conduct are further victimized.

Your concern, it seems to me, is that you don't want to have groups of people making decisions in private without accountability.  Transparency has been proven to be beneficial in many sorts of decision-making processes in Node.js, and you have been at the forefront of a lot of that work.  I can see that the suggestion of making process decisions in private upsets you.

I have also seen the dark side of closed-door process decisions, as you know.  There is a very good reason why I was among the people most vocally and consistently pushing for it being moved to an open governance model in a foundation.  The work that you and others in the TSC have done to make that a success is very admirable, and I think, a profound good.  I can relate with not wanting to see us back-slide down a slippery slope.  It's a scary proposition, one that might make you want to use distractingly colorful language such as references to secret religious torture brigades.  I can guess that you're probably feeling threatened and defensive about this, especially since it probably seems like you're alone in having this concern (at least in this thread/wg).

So, let me state that again: I am also concerned about losing openness.

However, let's not throw out this baby with the bathwater, by lionizing some narrow definition of "openness" without recognizing the real goal.  Being "open" (in the sense of "having conversations in public") is a means to an end.  Here are some other things that we should value, which *some* sorts of openness *usually* will tend towards (but probably not *all* types of openness *always*)

- regular predictable release cadence
- predictable supported lifetimes for releases
- well-tested, high-performant implementations of user-friendly APIs
- clear, well-organized, thorough documentation
- a welcoming community that encourages contribution and participation

These goals reinforce one another, and all in turn serve JIFASNIF.  Everyone involved in this WG values those things.  We all want node to be an easy way to write high-performance network programs.  If it was just about inclusiveness, we'd probably give up on node and go do this somewhere else.

If a person is targeted based on their inclusion in an underrepresented group, and victimized by ongoing harassment, then that is not fun.  That's not a welcoming community that encourages their participation and contribution.  They are not going to make the code faster, the docs clearer, or the tests testier.  And they're not going to report it in public, because that would put them in harm's way.  What's worse, other people will see that happening, and stay away.

Let's not value talking-in-public so much that we're willing to sacrifice inclusion though, because that means we do not have access to those people's time and attention, which could be used making the project better.

I want Node.js to be as open as possible, and that includes being open to receiving abuse reports in a way that makes sure that information is not lost, and open to people who are uncomfortable making certain sorts of statements in public.

If you still think I'm missing it, please try to restate it more clearly, and I'll do my best to understand.
